src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageEndpoint.cs

```diff
@@ -147,27 +143,39 @@ public async Task<RazorLanguageQueryResponse> Handle(RazorLanguageQueryParams re
             };
         }
 
-        public async Task<RazorMapToDocumentRangesResponse> Handle(RazorMapToDocumentRangesParams request, CancellationToken cancellationToken)
+        private async Task<DocumentSnapshotAndVersion?> TryGetDocumentSnapshotAndVersionAsync(string uri, CancellationToken cancellationToken)
+        {
+            return await _projectSnapshotManagerDispatcher.RunOnDispatcherThreadAsync<DocumentSnapshotAndVersion?>(() =>
```

Comment:
Nit: No need for the await here, can save a state machine

Reply:
If I do that it returns `ConfigurableAwaiter` instead of `Task` but I guess that's not the end of the world for a private method.

Reply:
You get rid of the `ConfigureAwait` call too.. you're not awaiting it, so you don't need to configure how its awaited